### ISSUE TITLE
db structure

### ISSUE THREAD

**Issue by yawnt** [spacedock-io/docker-index#1]
we need
- users
- repos
- images 
- tokens

repos have many images and many tokens
users have repos

**Comment by mmalecki** [comment]
So I was thinking an image per document? Does user belong to index or to both?

**Comment by yawnt** [comment]
all this stuff belongs to index, registry doesn't even need a database (i mean in our case it does since it's couchdb, but technically it wouldn't, it's just plain storage)

**Comment by Southern** [comment]
Users are only sending us a username, password, and email when they want to register. So that should be simple.

Repositories... not sure yet. Need to figure out what all is being sent to us. All I've seen hit so far is /v1/_ping when I tested it, but I don't think I tested it correctly looking back on it. I never did `docker tag` to add in a host like it says in the docs. So perhaps the error message I was seeing about the image not being found was my fault.

Images are files, aren't they? Shouldn't we be storing them somewhere other than couch?

Tokens, as in the ones sent back from the index to the registry... right? If that's the case, why can't we just save a list of them in the repo doc? One less query we have to make per user. Like:

``` js
repo.tokens = {
  'image id': 'token',
  'image2 id': 'token'
}
```
